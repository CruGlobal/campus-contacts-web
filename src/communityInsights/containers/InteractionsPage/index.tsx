import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';

import Card from '../../components/Card';
import { AppContext } from '../../../appContext';
import StagesSummary from '../../components/StagesSummary';
import ImpactInfo from '../../components/ImpactInfo';
import FiltersChart from '../../components/FiltersChart';

import {
    INTERACTIONS_TOTAL_REPORT,
    INTERACTIONS_TOTAL_COMPLETED_REPORT,
    INTERACTIONS_COMPLETED_REPORT,
} from './queries';
// INTERACTIONS_TOTAL_REPORT INTERFACES
import { impactReportInteractionsCount } from './__generated__/impactReportInteractionsCount';
// INTERACTIONS_TOTAL_COMPLETED_REPORT INTERFACES
import {
    communityReportInteractions,
    communityReportInteractions_community_report_interactions as communityReportInteractionsEntry,
} from './__generated__/communityReportInteractions';
// INTERACTIONS_COMPLETED_REPORT INTERFACES
import {
    communityReportDaysInteractions,
    communityReportDaysInteractions_community_report_daysReport as communityReportDaysInteractionsRow,
    communityReportDaysInteractions_community_report_daysReport_interactionResults as communityReportDaysInteractionsStage,
} from './__generated__/communityReportDaysInteractions';

const InteractionsPage = () => {
    const { t } = useTranslation('insights');
    const { orgId } = useContext(AppContext);

    return (
        <div>
            <ImpactInfo
                query={INTERACTIONS_TOTAL_REPORT}
                text={(report: impactReportInteractionsCount) =>
                    t('interactions.taken', {
                        count: report.community.impactReport.interactionsCount,
                        people:
                            report.community.impactReport
                                .interactionsReceiversCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card title={t('interactions.totalCompleted')}>
                <StagesSummary
                    query={INTERACTIONS_TOTAL_COMPLETED_REPORT}
                    variables={{ id: orgId }}
                    longNames={true}
                    mapData={(data: communityReportInteractions) =>
                        data.community.report.interactions.map(
                            (entry: communityReportInteractionsEntry) => ({
                                stage: entry.interactionType.name,
                                icon: entry.interactionType.name
                                    .toLowerCase()
                                    .split(' ')
                                    .join('-'),
                                count: entry.interactionCount,
                            }),
                        )
                    }
                />
            </Card>
            <Card
                title={t('interactions.completed')}
                subtitle={t('interactions.completedSubtitle')}
            >
                <FiltersChart
                    nullContent={'interactionsCompleted'}
                    query={INTERACTIONS_COMPLETED_REPORT}
                    variables={{ id: orgId }}
                    mapData={(data: communityReportDaysInteractions) =>
                        data.community.report.daysReport.map(
                            (row: communityReportDaysInteractionsRow) => ({
                                ['total']: row.interactionsCount,
                                ['stages']: row.interactionResults.map(
                                    (
                                        stage: communityReportDaysInteractionsStage,
                                    ) => ({
                                        name: stage.interactionType.name,
                                        count: stage.count,
                                    }),
                                ),
                                ['date']: row.date,
                            }),
                        )
                    }
                    currentDate={moment().toDate()}
                    label={t('interactions.legend')}
                />
            </Card>
        </div>
    );
};

export default InteractionsPage;
