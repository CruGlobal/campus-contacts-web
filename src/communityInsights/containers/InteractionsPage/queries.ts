import gql from 'graphql-tag';

const INTERACTIONS_TOTAL_REPORT = gql`
    query impactReportInteractionsCount($id: ID!) {
        community(id: $id) {
            impactReport {
                interactionsCount
                interactionsReceiversCount
            }
        }
    }
`;

const INTERACTIONS_TOTAL_COMPLETED_REPORT = gql`
    query communityReportInteractions(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                interactions {
                    nodes {
                        interactionCount
                        interactionType {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const INTERACTIONS_COMPLETED_REPORT = gql`
    query communityReportDaysInteractions(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        interactionsCount
                        interactionResults {
                            nodes {
                                count
                                interactionType {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

export {
    INTERACTIONS_TOTAL_REPORT,
    INTERACTIONS_TOTAL_COMPLETED_REPORT,
    INTERACTIONS_COMPLETED_REPORT,
};
