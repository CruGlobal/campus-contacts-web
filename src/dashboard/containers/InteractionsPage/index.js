import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';
import _ from 'lodash';

import Card from '../../components/Card';
import AppContext from '../../appContext';
import StagesSummary from '../../components/StagesSummary';
import ImpactInfo from '../../components/ImpactInfo';
import FiltersChart from '../../components/FiltersChart';

import {
    INTERACTIONS_TOTAL_REPORT,
    INTERACTIONS_TOTAL_COMPLETED_REPORT,
    INTERACTIONS_COMPLETED_REPORT,
} from './queries';

const InteractionsPage = () => {
    const { t } = useTranslation('insights');
    const { orgId } = useContext(AppContext);

    return (
        <div>
            <ImpactInfo
                query={INTERACTIONS_TOTAL_REPORT}
                text={report =>
                    t('interactions.taken', {
                        count:
                            report.communityReport.impactReport
                                .interactionsCount,
                        people:
                            report.communityReport.impactReport
                                .interactionsReceiversCount,
                    })
                }
                variables={{ organizationId: orgId }}
            />
            <Card title={t('interactions.totalCompleted')}>
                <StagesSummary
                    query={INTERACTIONS_TOTAL_COMPLETED_REPORT}
                    variables={{
                        organizationId: orgId,
                    }}
                    mapData={data =>
                        data.communityReport.interactions.map(entry => ({
                            stage: entry.interactionType.name,
                            icon: entry.interactionType.name
                                .toLowerCase()
                                .split(' ')
                                .join('-'),
                            count: entry.interactionCount,
                        }))
                    }
                />
            </Card>
            <Card
                title={t('interactions.completed')}
                subtitle={t('interactions.completedSubtitle')}
            >
                <FiltersChart
                    query={INTERACTIONS_COMPLETED_REPORT}
                    mapData={data =>
                        data.communityReport.dayReport.map(row => ({
                            ['total']: row.interactionCount,
                            ['stages']: row.interactions.map(stage => ({
                                name: stage.interactionType.name,
                                count: stage.interactionCount,
                            })),
                            ['date']: row.date.substring(0, 2),
                        }))
                    }
                    countAverage={data =>
                        Math.floor(_.sumBy(data, 'total') / data.length)
                    }
                    label={t('interactions.legend')}
                />
            </Card>
        </div>
    );
};

export default InteractionsPage;

InteractionsPage.propTypes = {};
