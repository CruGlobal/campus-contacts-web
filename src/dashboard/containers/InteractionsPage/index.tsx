import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';

import Card from '../../components/Card';
import AppContext from '../../appContext';
import StagesSummary from '../../components/StagesSummary';
import ImpactInfo from '../../components/ImpactInfo';
import FiltersChart from '../../components/FiltersChart';

import {
    INTERACTIONS_TOTAL_REPORT,
    INTERACTIONS_TOTAL_COMPLETED_REPORT,
    INTERACTIONS_COMPLETED_REPORT,
} from './queries';

const InteractionsPage = () => {
    const { t } = useTranslation('insights');
    const { orgId }: any = useContext(AppContext);

    return (
        <div>
            <ImpactInfo
                query={INTERACTIONS_TOTAL_REPORT}
                text={report =>
                    t('interactions.taken', {
                        count: report.community.impactReport.interactionsCount,
                        people:
                            report.community.impactReport
                                .interactionsReceiversCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card title={t('interactions.totalCompleted')}>
                <StagesSummary
                    query={INTERACTIONS_TOTAL_COMPLETED_REPORT}
                    variables={{ id: orgId }}
                    mapData={data =>
                        data.community.report.interactions.map(
                            (entry: any) => ({
                                stage: entry.interactionType.name,
                                icon: entry.interactionType.name
                                    .toLowerCase()
                                    .split(' ')
                                    .join('-'),
                                count: entry.interactionCount,
                            }),
                        )
                    }
                />
            </Card>
            <Card
                title={t('interactions.completed')}
                subtitle={t('interactions.completedSubtitle')}
            >
                <FiltersChart
                    query={INTERACTIONS_COMPLETED_REPORT}
                    variables={{ id: orgId }}
                    mapData={data =>
                        data.community.report.daysReport.map((row: any) => ({
                            ['total']: row.interactionsCount,
                            ['stages']: row.interactionResults.map(
                                (stage: any) => ({
                                    name: stage.interactionType.name,
                                    count: stage.count,
                                }),
                            ),
                            ['date']: row.date,
                        }))
                    }
                    currentDate={moment().toDate()}
                    label={t('interactions.legend')}
                />
            </Card>
        </div>
    );
};

export default InteractionsPage;

InteractionsPage.propTypes = {};
