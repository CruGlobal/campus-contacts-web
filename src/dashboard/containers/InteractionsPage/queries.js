import gql from 'graphql-tag';

const INTERACTIONS_TOTAL_REPORT = gql`
    query communityReport {
        communityReport {
            impactReport {
                interactionsCount
                interactionsReceiversCount
            }
        }
    }
`;

const INTERACTIONS_TOTAL_COMPLETED_REPORT = gql`
    query communityReport {
        communityReport {
            interactions {
                interactionCount
                interactionType {
                    name
                }
            }
        }
    }
`;

const INTERACTIONS_COMPLETED_REPORT = gql`
    query communityReport {
        communityReport {
            dayReport {
                date
                interactionCount
                interactions {
                    interactionCount
                    interactionType {
                        name
                    }
                }
            }
        }
    }
`;

export {
    INTERACTIONS_TOTAL_REPORT,
    INTERACTIONS_TOTAL_COMPLETED_REPORT,
    INTERACTIONS_COMPLETED_REPORT,
};
