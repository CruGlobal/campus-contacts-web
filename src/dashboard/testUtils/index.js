import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
