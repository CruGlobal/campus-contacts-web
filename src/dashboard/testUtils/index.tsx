import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
