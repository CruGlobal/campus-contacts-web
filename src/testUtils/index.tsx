import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
