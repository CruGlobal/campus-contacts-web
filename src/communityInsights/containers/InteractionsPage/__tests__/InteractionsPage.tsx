import React from 'react';

import { renderWithContext } from '../../../../testUtils';

import InteractionsPage from '..';

describe('<PersonalStepsPage />', () => {
    it('should render properly in loading state', async () => {
        const { snapshot, unmount } = renderWithContext(<InteractionsPage />, {
            appContext: {
                orgId: '1',
            },
        });
        snapshot();
        unmount();
    });

    // TODO: Add a test with mocking data when GraphQL schema contains above query
});
