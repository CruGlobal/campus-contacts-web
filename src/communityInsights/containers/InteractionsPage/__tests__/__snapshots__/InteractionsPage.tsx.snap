// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalStepsPage /> should render properly in loading state 1`] = `
.emotion-0 {
  font-weight: 300;
  font-size: 30px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 0px;
  margin-top: 48px;
  padding-left: 21px;
}

.emotion-5 {
  box-shadow: 0px 2px 12px rgba(0,0,0,0.2);
  background: #ffffff;
  border-radius: 8px;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 43px;
  margin-top: 36px;
}

.emotion-1 {
  font-weight: 300;
  font-size: 36px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 12px;
  margin-top: 0px;
}

.emotion-2 {
  font-size: 14px;
  line-height: 20px;
  font-weight: normal;
  color: #505256;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-4 {
  padding: 25px;
  margin: 0;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div>
    <h1
      class="emotion-0"
    >
      <span>
        Gathering your stats...
      </span>
    </h1>
    <div
      class="emotion-5"
    >
      <h1
        class="emotion-1"
      >
        Total Interactions Completed
      </h1>
      <h2
        class="emotion-2"
      />
      <div
        class="emotion-4"
      >
        <div
          class="emotion-3"
        >
          Loading...
        </div>
      </div>
    </div>
    <div
      class="emotion-5"
    >
      <h1
        class="emotion-1"
      >
        Interactions Completed
      </h1>
      <h2
        class="emotion-2"
      >
        Total interactions completed over time.
      </h2>
      <div
        class="emotion-4"
      >
        <div>
          Loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;
